### DISCUSSION TITLE
Issue while loging in the client machine on the same network

### DISCUSSION THREAD

**Discussion by vskmizoram** [Sunbird-cQube/community#855] [Q&A]
I have installed the Cqube Ver 5.0.3 on a server.  In the server machine, I can log in using the username and password for my dashboard. But while logging in with the same username and password in the client machine on the same local network it says the invalid credentials. Could you help me out if any change is required in the setting of the keycloak module? I can access the keycloak from the server and the client machine.

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Can you please send me the URL of  mizoram ?

**Reply by vskmizoram** [reply]
We have installed it on localhost, we can access the dashboard on the Server using 192.168.1.9 (local IP) but if we are using the same in the client machine (in the same local network using IP 192.168.1.9) then the dashboard appears but after entering the credentials it says " invalid credentials"

**Reply by bommalla** [reply]
@vskmizoram @RNarmatha 

Please check the below localhost deployment document and check the all the port numbers are enabled or not ?

https://docs.google.com/document/d/1ZSpf6AX214GSbL4RvbTwTwVAJbIXbMjdTd5lNBOkUQc/edit#heading=h.7chf9swggyzp

**Reply by vskmizoram** [reply]
Need access to vskmizoram@gmail.com to access the document

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Done please check now.

**Reply by vskmizoram** [reply]
I had used the same document for installation. It was working on localhost successfully

**Reply by vskmizoram** [reply]
![Screenshot from 2024-01-25 13-43-42](https://github.com/Sunbird-cQube/community/assets/157444637/377bc3f0-39c5-40d9-83ac-04682f3a8183)

**Reply by vskmizoram** [reply]
But I am not able to open in client machine using address 192.168.1.9:4200... the dashboard is coming using this address but when i enter the credentials , it says "invalid credentials"

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

If your accessing both servers in same internet it will work.else won't be access it. better we can install cQube with proper domain name.

**Reply by vskmizoram** [reply]
I am not talking about the internet. I am talking about the intranet using local IPs within the organisation which does not required the internet. under Intranet the all client machine and server communicate with each other through the local IPs assigned to them. So , I need to access on the local client machine which is directly connected to the server

**Reply by vskmizoram** [reply]
You can call me on 7827657230 so that I can better explain you my issue

**Comment by RNarmatha** [comment]
@bommalla -please schedule one call with them and share the invite here

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Please tell me your availability then I can schedule call accordingly.

**Reply by vskmizoram** [reply]
any time you can connect us

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 
Please find the below meeting details 

VSK mizoram issue
Thursday, January 25 · 2:30 – 3:00pm
Time zone: Asia/Kolkata
Google Meet joining info
Video call link: https://meet.google.com/gcn-tpnt-vmk
Or dial: ‪(US) +1 530-552-1556‬ PIN: ‪388 424 484‬#
More phone numbers: https://tel.meet/gcn-tpnt-vmk?pin=1124077378843

**Reply by vskmizoram** [reply]
I have created one user name vsk in  Keycloak and try to access the dashboard from client machine, but again it was saying "invalid credentials" the keycloak log file is being attached for your reference as suggested
[err.log](https://github.com/Sunbird-cQube/community/files/14050090/err.log)

**Reply by vskmizoram** [reply]
but if i connecting with the new user in the server machine it is giving me the data dashboard after entering the credentials

**Reply by vskmizoram** [reply]
now keycloak is working in the client machine.. can we connect now

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Let me check the logs then we will connect.

**Reply by vskmizoram** [reply]
The log might contain the successful login with vsk user.. but this I have tried on the Server machine not on the client. The login failed details shown in the log for vsk is from the client machine

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Please follow the below steps to rebuild the key-cloak 

1. First go to below mentioned path 
cd cqube-devopes/microservices

2. Using blow commands to rebuild the key-cloak container 

sudo docker stop keycloak_app
sudo docker rm keycloak_app
sudo docker rmi keycloak_ms:1

sudo docker build -t keycloak_ms:1 .

3. Then navigate the below path cqube_devops/ansible and run below command
sudo docker-compose -f docker-compose.yml up -d

4. Then navigate the below path and run the keycloak.sh file
cd cqube_devops/ansible
sudo bash shell-scripts/keycloak.sh

**Reply by vskmizoram** [reply]
facing issue in building the keycloak_ms:1 .

the following error is coming

![Uploading image.png…]()

**Reply by vskmizoram** [reply]
![image](https://github.com/Sunbird-cQube/community/assets/157444637/e5b31863-c727-49dd-8f61-ce860d412fa2)

**Reply by vskmizoram** [reply]
sudo docker build -t keycloak_ms:1 . is not working

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Please run the above command in below mentioned path

cd /cqube-devops/microservices/keycloak-ms

sudo docker build -t keycloak_ms:1 .

**Reply by vskmizoram** [reply]
I have successfully done all the procedures.. now Keycloak has opened..., and the new user have been created .. but still I am not able to access the dashboard from the client machine, the login page is coming.. but it was saying "Invalid Credentials"

**Reply by vskmizoram** [reply]
<img width="616" alt="image" src="https://github.com/Sunbird-cQube/community/assets/157444637/85f2852f-f818-4b23-a629-8823551cd5d7">

this is coming on client's machine

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

I'll check and get back to you.give me some time

**Reply by vskmizoram** [reply]
waiting for your response

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Actually localhost deployment is used for only single laptop deployment. we haven't test this kind of approach.

First I need check this kind of approach with my devops team.

**Reply by vskmizoram** [reply]
The localhost means the intranet deployment, it should be available to all clients connected with the same network in the intranet.. We are implementing in our State Data Server available in office premises. The local network has been established and is connected to the VSK Server. The localhost is accessible from each client and the VSK login page is available. But while entering the credentials it is floating error "Invalid Credentials"...

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

we haven't worked  like this kind  of scenario. we have to check with the team and take it up this task . meanwhile it could be great if you drop an email on the same to Rahul (who is the product owner of cQube)

**Reply by vskmizoram** [reply]
But some solution needs to be there of this type of problem most of the states use the cloud for VSK but at Mizoram, we are not going for a cloud , we are preparing our own State Data Center at our primeses so we have to check on our local intranet

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

I spoke with my devops team regarding this particular issue.They are not tested this kind of approach with localhost. Once my devops team available I can schedule call and discuss on this..

**Reply by vskmizoram** [reply]
One thing to clear, I am not using the "localhost" word from my browser.. I am using "192.168.1.9" IP address to access the server. The Login page is coming ... see below 

<img width="1261" alt="image" src="https://github.com/Sunbird-cQube/community/assets/157444637/d4e80842-e3b2-4fd9-bc4b-a5bd2fd67b8b">

but while entering the credentials it says "Invalid Credentials" when I inspect this page .. it goes to http://localhost:3002/login for verification ..

**Reply by vskmizoram** [reply]
I am eagerly waiting for your response regarding connectivity from the client machine to the server using the local private IP

**Reply by RNarmatha** [reply]
Hi @vskmizoram, 
Good Morning!  
The team is currently occupied with sprint release activities. Rajesh will get back to you once his work is completed. 
Thank you for your cooperation .

Regards 
Narmatha

**Reply by vskmizoram** [reply]
Thanx.. can you please provide the Rajesh Number on vskmizoram@gmail.com

**Reply by RNarmatha** [reply]
Hi @vskmizoram , Tomorrow Rajesh will schedule a with you and our Devops team . In that call we can discuss more in detail. 

Thanks for your cooperation 
Narmatha

**Reply by RNarmatha** [reply]
Please let me know your availability tomorrow

**Reply by vskmizoram** [reply]
We can connect at 2:00 pm

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

If you possible can we connect at 12 PM today ?

**Reply by vskmizoram** [reply]
Yes we can connect

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

Okey,Thanks

**Comment by bommalla** [comment]
@vskmizoram @RNarmatha 

I scheduled  call at 12 PM today. Please be available at the time.

Mizoram deployment issue
Thursday, February 1 · 12:00 – 1:00pm
Time zone: Asia/Kolkata
Google Meet joining info
Video call link: https://meet.google.com/deh-wybx-xaj
Or dial: ‪(US) +1 978-347-2454‬ PIN: ‪310 407 657‬#
More phone numbers: https://tel.meet/deh-wybx-xaj?pin=4616469160642

**Comment by bommalla** [comment]
@vskmizoram 

Are you available for call at 3PM ?  with my UI developer  ?

**Reply by vskmizoram** [reply]
yes..

**Reply by bommalla** [reply]
Thanks.I'm scheduling call at 3PM
